components/core/src/clp/streaming_archive/reader/Archive.hpp

```diff
@@ -128,7 +145,8 @@ class Archive {
             std::string const& file_path,
             segment_id_t segment_id
     ) {
-        return m_metadata_db.get_file_iterator(begin_ts, end_ts, file_path, true, segment_id);
+        return m_metadata_db
```

Comment:
Let's also add `order_by_segment_end_ts` as an argument here since callers may want to do so.